apps/meteor/client/lib/chats/uploads.ts

```diff
@@ -128,6 +127,9 @@ const send = async (
 						t,
 						content,
 					});
+				} else if (xhr.readyState === xhr.DONE && xhr.status === 400) {
```

Comment:
WYT about this approach?

```tsx
xhr.onload = async () => {
	if (xhr.readyState === xhr.DONE) {
		if (xhr.status === 400) {
			const error = JSON.parse(xhr.responseText);
			updateUploads((uploads) => [...uploads, { ...upload, error: new Error(error.error) }]);
			return;
		}
	
		if (xhr.status === 200) {
			const result = JSON.parse(xhr.responseText);
			let content;
			if (getContent) {
				content = await getContent(result.file._id, result.file.url);
			}
	
			await sdk.rest.post(`/v1/rooms.mediaConfirm/${rid}/${result.file._id}`, {
				msg,
				tmid,
				description,
				t,
				content,
			});
		}
	}
};
```

Reply:
Wanted to minimize the diff, but I committed your suggestion. Should be the same. 👍